onboarding_project/squad_pantry_app/admin.py

```diff
@@ -22,11 +25,87 @@ class OrderDishInline(admin.StackedInline):
     formset = BaseOrderDishFormset
     extra = 1
 
+    def formfield_for_foreignkey(self, db_field, request, **kwargs):
+        if db_field.name == "dish":
+            kwargs["queryset"] = Dish.objects.filter(is_available=True)
+        return super().formfield_for_foreignkey(db_field, request, **kwargs)
+
+    def get_readonly_fields(self, request, obj=None):
+        try:
+            if obj.placed_by:
+                return self.readonly_fields + ('dish', 'quantity')
+        except AttributeError:
```

Comment:
Because this is a kind of hack. We need to make them `readonly` if we are reading the order and not creating it for the first time. So it throws an `AttributeError` because initially `placed_by` is not assigned. `placed_by` is assigned after the user clicks on `save` for creating the order.

Reply:
So we need to check if the order has been already created or not here. There's a better way to do that without catching `AttributeError`. Think/Google about it.

Reply:
Done!